handlers/preview.go

```diff
@@ -157,7 +157,15 @@ func (f *Filter) OutputPage() http.HandlerFunc {
 
 		p := mapper.CreatePreviewPage(req, dimensions, fj, dataset, filterOutputID, datasetID, ver.ReleaseDate, f.APIRouterVersion, f.EnableDatasetPreview, lang)
 
-		if latestPath == versionPath {
+		editionDetails, err := f.DatasetClient.GetEdition(req.Context(), userAccessToken, "", collectionID, datasetID, edition)
+		if err != nil {
+			log.Event(ctx, "failed to get edition details", log.ERROR, log.Error(err), log.Data{"dataset": datasetID, "edition": edition})
+			setStatusCode(req, w, err)
+			return
+		}
+
+		latestVersionInEditionPath := fmt.Sprintf("/datasets/%s/editions/%s/versions/%s", datasetID, edition, editionDetails.Links.LatestVersion.ID)
```

Comment:
I was debating whether or not the logic here (l160-167) could be separated out into its own function within the handlers package, but it doesn't seem like something we do elsewhere, so I've left it within each of the handler funcs

Reply:
I don't see the benefit of separating this into a new function since the new function would only contain the `GetEdition` call and then the additional line (`latestVersionInEditionPath`). Furthermore, in the `update-version.go`, the `GetEdition` function is called and it isn't used for `latestVersionInEditionPath` in which the `GetEdition` is used differently. By not combining it, it is more clearer to see what is happening and how the value is passed in the package. Also, generally creating a function to use `GetEdition` with one additional line feels unnecessary.